tfx/components/example_gen/utils.py

```diff
@@ -335,37 +363,68 @@ def _create_matching_glob_and_regex(
   if is_match_span:
     # Check if span spec has any width args. Defaults to greedy matching if
     # no width modifiers are present.
-    span_width_regex = '.*'
+    span_glob_replace = '*'
+    span_regex_replace = '.*'
     result = re.search(SPAN_FULL_REGEX, split.pattern)
     assert result, 'No Span found in Split %s' % split.pattern
     span_width_str = result.group('width')
     if span_width_str:
       try:
-        if int(span_width_str) <= 0:
+        span_width_int = int(span_width_str)
+        if span_width_int <= 0:
           raise ValueError('Not a positive integer.')
-        span_width_regex = '.{%s}' % span_width_str
+        span_regex_replace = '.{%s}' % span_width_str
       except ValueError:
         raise ValueError(
             'Width modifier in span spec is not a positive integer: %s' %
             split.pattern)
 
-    split_glob_pattern = re.sub(SPAN_FULL_REGEX, '*', split_glob_pattern)
+    if range_config and range_config.HasField('static_range'):
+      # If using RangeConfig.static_range, replace span spec in patterns
+      # with given span from static range.
+      span_str = str(range_config.static_range.start_span_number)
+      if span_width_str:
+        span_width_int = int(span_width_str)
+        if span_width_int < len(span_str):
+          raise ValueError(
+              'Span spec width is less than number of digits in span: (%s, %s)'
+              % (span_width_int, span_str))
+        span_str = span_str.zfill(span_width_int)
+
+      span_regex_replace = span_str
+      span_glob_replace = span_str
```

Comment:
consider to move line 382 - 395, line 411 - 419 to a util, e.g., _replace_split_glob_and_regex_, which helps the reader to grasp the overview 

in addition, this util can be used in the callsite, e.g., _create_matching_glob_and_regex, then 
```
_create_matching_glob_and_regex
if range_config:
   _replace_split_glob_and_regex(range-config)
```

Reply:
the main issue i see with using this in the call site is that when we create the glob pattern for a pattern say `span{SPAN}/ver{VERSION}/*` the resulting glob is `span*/ver*/*`. There doesn't seem to be a clean way to identify which asterisk was originally the span tag and which was the version tag, which would make subsequent substitution in the glob a bit messy. Was there a different logic you had in mind for this callsite usage?

Reply:
refactored out width into a seperate method as well as range_config replacement logic